### ISSUE TITLE
Cannot use CellSize in dS integral with Hermite element

### ISSUE THREAD

**Issue by pefarrell** [firedrakeproject/firedrake#1528]
Consider the following code:

```
from firedrake import *

mesh = UnitIntervalMesh(100)
V = FunctionSpace(mesh, "Hermite", 3)

u = Function(V)
v = TestFunction(V)

h = CellSize(mesh)

F = (
      inner(grad(u), grad(v))*dx
    - inner(Constant(1), v)*dx
    + avg(h) * inner(jump(grad(u)), jump(grad(v)))*dS
    )

bc = DirichletBC(V, 0, "on_boundary")
solver_parameters = {"snes_type": "newtonls",
                     "snes_monitor": None,
                     "snes_linesearch_type": "basic",
                     "ksp_type": "preonly",
                     "pc_type": "lu",
                     "pc_factor_mat_solver_type": "umfpack"}

solve(F == 0, u, bc, solver_parameters=solver_parameters)
```

The example is a little contrived because it's the reduction of something else, but no matter. I expected it to compile, but when I run this it fails with

```
Traceback (most recent call last):
  File "/mnt/ntfs/data/install-scripts/firedrake/firedrake-dev-20190909-mpich/src/PyOP2/pyop2/caching.py", line 197, in __new__
    return cls._cache_lookup(key)
  File "/mnt/ntfs/data/install-scripts/firedrake/firedrake-dev-20190909-mpich/src/firedrake/firedrake/tsfc_interface.py", line 59, in _cache_lookup
    return cls._cache.get(key) or cls._read_from_disk(key, comm)
  File "/mnt/ntfs/data/install-scripts/firedrake/firedrake-dev-20190909-mpich/src/firedrake/firedrake/tsfc_interface.py", line 80, in _read_from_disk
    raise KeyError("Object with key %s not found" % key)
KeyError: 'Object with key bf0fe7f15f233439b0e192f5e543a283 not found'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "hermite-cellsize.py", line 25, in <module>
    solve(F == 0, u, bc, solver_parameters=solver_parameters)
  File "/mnt/ntfs/data/install-scripts/firedrake/firedrake-dev-20190909-mpich/src/firedrake/firedrake/solving.py", line 125, in solve
    _solve_varproblem(*args, **kwargs)
  File "/mnt/ntfs/data/install-scripts/firedrake/firedrake-dev-20190909-mpich/src/firedrake/firedrake/solving.py", line 168, in _solve_varproblem
    appctx=appctx)
  File "/mnt/ntfs/data/install-scripts/firedrake/firedrake-dev-20190909-mpich/src/firedrake/firedrake/variational_solver.py", line 182, in __init__
    options_prefix=self.options_prefix)
  File "/mnt/ntfs/data/install-scripts/firedrake/firedrake-dev-20190909-mpich/src/firedrake/firedrake/solving_utils.py", line 131, in __init__
    form_compiler_parameters=self.fcp)
  File "/mnt/ntfs/data/install-scripts/firedrake/firedrake-dev-20190909-mpich/src/firedrake/firedrake/assemble.py", line 158, in create_assembly_callable
    loops = tuple(loops)
  File "/mnt/ntfs/data/install-scripts/firedrake/firedrake-dev-20190909-mpich/src/firedrake/firedrake/assemble.py", line 228, in _assemble
    kernels = tsfc_interface.compile_form(f, "form", parameters=form_compiler_parameters, inverse=inverse)
  File "/mnt/ntfs/data/install-scripts/firedrake/firedrake-dev-20190909-mpich/src/firedrake/firedrake/tsfc_interface.py", line 218, in compile_form
    number_map, interface, coffee).kernels
  File "/mnt/ntfs/data/install-scripts/firedrake/firedrake-dev-20190909-mpich/src/PyOP2/pyop2/caching.py", line 199, in __new__
    obj = make_obj()
  File "/mnt/ntfs/data/install-scripts/firedrake/firedrake-dev-20190909-mpich/src/PyOP2/pyop2/caching.py", line 189, in make_obj
    obj.__init__(*args, **kwargs)
  File "/mnt/ntfs/data/install-scripts/firedrake/firedrake-dev-20190909-mpich/src/firedrake/firedrake/tsfc_interface.py", line 127, in __init__
    tree = tsfc_compile_form(form, prefix=name, parameters=parameters, interface=interface, coffee=coffee)
  File "/mnt/ntfs/data/install-scripts/firedrake/firedrake-dev-20190909-mpich/src/tsfc/tsfc/driver.py", line 63, in compile_form
    kernel = compile_integral(integral_data, fd, prefix, parameters, interface=interface, coffee=coffee)
  File "/mnt/ntfs/data/install-scripts/firedrake/firedrake-dev-20190909-mpich/src/tsfc/tsfc/driver.py", line 197, in compile_integral
    **config)
  File "/mnt/ntfs/data/install-scripts/firedrake/firedrake-dev-20190909-mpich/src/tsfc/tsfc/fem.py", line 645, in compile_ufl
    result = map_expr_dags(context.translator, expressions)
  File "/mnt/ntfs/data/install-scripts/firedrake/firedrake-dev-20190909-mpich/src/ufl/ufl/corealg/map_dag.py", line 84, in map_expr_dags
    r = handlers[v._ufl_typecode_](v)
  File "/mnt/ntfs/data/install-scripts/firedrake/firedrake-dev-20190909-mpich/src/tsfc/tsfc/ufl_utils.py", line 144, in _modified_terminal
    return self.modified_terminal(o)
  File "/mnt/ntfs/data/install-scripts/firedrake/firedrake-dev-20190909-mpich/src/tsfc/tsfc/fem.py", line 303, in modified_terminal
    return translate(mt.terminal, mt, self.context)
  File "/usr/lib/python3.6/functools.py", line 807, in wrapper
    return dispatch(args[0].__class__)(*args, **kw)
  File "/mnt/ntfs/data/install-scripts/firedrake/firedrake-dev-20190909-mpich/src/tsfc/tsfc/fem.py", line 564, in translate_argument
    table = ctx.entity_selector(callback, mt.restriction)
  File "/mnt/ntfs/data/install-scripts/firedrake/firedrake-dev-20190909-mpich/src/tsfc/tsfc/fem.py", line 101, in entity_selector
    return gem.select_expression(list(map(callback, self.entity_ids)), f)
  File "/mnt/ntfs/data/install-scripts/firedrake/firedrake-dev-20190909-mpich/src/tsfc/tsfc/fem.py", line 552, in callback
    finat_dict = ctx.basis_evaluation(element, mt, entity_id)
  File "/mnt/ntfs/data/install-scripts/firedrake/firedrake-dev-20190909-mpich/src/tsfc/tsfc/fem.py", line 228, in basis_evaluation
    coordinate_mapping=CoordinateMapping(mt, self))
  File "/mnt/ntfs/data/install-scripts/firedrake/firedrake-dev-20190909-mpich/src/FInAT/finat/physically_mapped.py", line 95, in basis_evaluation
    M = self.basis_transformation(coordinate_mapping)
  File "/mnt/ntfs/data/install-scripts/firedrake/firedrake-dev-20190909-mpich/src/FInAT/finat/hermite.py", line 22, in basis_transformation
    h = coordinate_mapping.cell_size()
  File "/mnt/ntfs/data/install-scripts/firedrake/firedrake-dev-20190909-mpich/src/tsfc/tsfc/fem.py", line 138, in cell_size
    return self.interface.cell_size(self.mt.restriction)
  File "/mnt/ntfs/data/install-scripts/firedrake/firedrake-dev-20190909-mpich/src/tsfc/tsfc/kernel_interface/common.py", line 67, in cell_size
    return gem.partial_indexed(self._cell_sizes, f)
  File "/mnt/ntfs/data/install-scripts/firedrake/firedrake-dev-20190909-mpich/src/tsfc/gem/gem.py", line 814, in partial_indexed
    elif len(indices) < len(tensor.shape):
AttributeError: 'tuple' object has no attribute 'shape'
```

**Comment by wence-** [comment]
Can you try that tsfc branch?

**Comment by danshapero** [comment]
I just ran this now and it seems to have worked, although I replaced UMFPACK with MUMPS:

    0 SNES Function norm 9.950572289516e-02 
    1 SNES Function norm 2.484806151041e-14

Safe to close?

**Comment by pefarrell** [comment]
Yep, thanks!

**Comment by wence-** [comment]
Oh yeah, I fixed this in firedrakeproject/tsfc#205.
